site-src/geps/gep-724.md

```diff
@@ -0,0 +1,426 @@
+  // route. This field is ignored for other values of "Allow".
+  //
+  // +optional
+  GatewayRefs []GatewayReference `json:"gatewayRefs,omitempty"`
+}
+```
+
+#### Added
+To Route Specs:
+```go
+  // AttachTo defines the parent resources this Route should be attached to. The
+  // only kind of target resource with "Core" support is Gateway. This API may
+  // be extended in the future to support additional kinds of parent resources
+  // such as Routes.
+  //
+  // +optional
+  // +kubebuilder:validation:MaxItems=16
+  AttachTo []AttachRef `json:"gateways,omitempty"`
+```
+
+And the struct that references:
+```go
+// AttachRef identifies an API object to attach to. The only kind of target
+// resource with "Core" support is Gateway. This API may be extended in the
+// future to support additional kinds of parent resources such as Routes.
+type AttachRef struct {
+  // Group is the group of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +kubebuilder:default=networking.x-k8s.io
+  Group string `json:"group"`
+
+  // Kind is kind of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Kind string `json:"kind"`
+
+  // Name is the name of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Name string `json:"name"`
+
+  // SectionName is the name of a section within the target resource. When
+  // unspecified, this targets the entire resource. In the following
+  // resources, SectionName is interpreted as the following:
+  // * Gateway: Listener Name
+  // * Route: Rule Name
+  //
+  // Support: Core
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +optional
+  SectionName string `json:"sectionName,omitempty"`
+
+  // Namespace is the namespace of the referent. When unspecified, the local
+  // namespace is inferred unless targeting a cluster-scoped resource in which
+  // case no namespace is inferred.
+  //
+  // Support: Extended
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +optional
+  Namespace string `json:"namespace,omitempty"`
+}
+```
+
+### Advantages
+
+* Simplifies the API by providing a single way to attach Routes to Gateway.
+* Assigns clear responsibilities to Gateway and Route owners.
+* Consistent with pattern of direct references from Routes to all associated
+  resources.
+* Enables attaching Routes to arbitrary parents, such as custom Gateways, other
+  Routes (to be defined), or Meshes.
+* Prevents accidental exposure of Routes.
+* Easy to understand which Gateways/parents a Route is attached to.
+* Further simplifies path to Route inclusion.
+* Follows pattern of direct reference in one direction with a broader trust
+  reference in the other direction.
+* Aligns with initial user feedback.
+
+### Disadvantages
```

Comment:
The easy path becomes complex since Routes are now Gateway aware.

Reply:
What is the easy path here? The default of a Gateway selecting all Routes in a namespace?

Reply:
Yes, a route allowing any gateways to bind to itself.

Reply:
Although those may be simpler, I think they're poor/insecure defaults. I'd prefer to move away from them in favor of more explicit attachment.

Reply:
Added something in the new rationale section that covers this.